vector/src/main/java/im/vector/app/features/createdirect/CreateDirectRoomByQrCodeFragment.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.createdirect
+
+import android.widget.Toast
+import androidx.annotation.CallSuper
+import com.airbnb.mvrx.activityViewModel
+import com.google.zxing.Result
+import com.google.zxing.ResultMetadataType
+import im.vector.app.R
+import im.vector.app.core.di.ScreenComponent
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.features.userdirectory.PendingInvitee
+import kotlinx.android.synthetic.main.fragment_qr_code_scanner.*
+import me.dm7.barcodescanner.zxing.ZXingScannerView
+import org.matrix.android.sdk.api.session.Session
+import org.matrix.android.sdk.api.session.permalinks.PermalinkData
+import org.matrix.android.sdk.api.session.permalinks.PermalinkParser
+import org.matrix.android.sdk.api.session.user.model.User
+import javax.inject.Inject
+
+class CreateDirectRoomByQrCodeFragment @Inject constructor() : VectorBaseFragment(), ZXingScannerView.ResultHandler {
+
+    @Inject lateinit var session: Session
+
+    override fun getLayoutResId() = R.layout.fragment_qr_code_scanner
+
+    private val createDirectRoomViewModel: CreateDirectRoomViewModel by activityViewModel()
+
+    @CallSuper
+    override fun injectWith(injector: ScreenComponent) {
+        session = injector.activeSessionHolder().getActiveSession()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        // Register ourselves as a handler for scan results.
+        scannerView.setResultHandler(this)
+        // Start camera on resume
+        scannerView.startCamera()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        // Stop camera on pause
+        scannerView.stopCamera()
```

Comment:
Shouldn't we call `scannerView.setResultHandler(null)` for symmetry?

Reply:
I have no idea :P but I changed it